common/achievement/util.ts

```diff
@@ -24,26 +24,48 @@ export function getMetricsByAction<ID extends ActionID>(actionId: ID): Metric[]
     return metricsByAction.get(actionId) || [];
 }
 
-export function getRelationTypeAndId(relation: string): [type: RelationTypes, id: number] {
+export function getRelationTypeAndId(relation: string): [type: RelationTypes, id: string] {
     const validRelationTypes = ['match', 'subcourse'];
     const [relationType, id] = relation.split('/');
     if (!validRelationTypes.includes(relationType)) {
         throw Error('No valid relation found in relation: ' + relationType);
     }
-    return [relationType as RelationTypes, Number(id)];
+    return [relationType as RelationTypes, id];
 }
 
-export async function getBucketContext(relation: string): Promise<AchievementContextType> {
-    const [type, id] = getRelationTypeAndId(relation);
+export async function getBucketContext(relation: string, id: string): Promise<AchievementContextType> {
+    const [userType, userId] = getUserTypeAndIdForUserId(id);
+    const [relationType, relationId] = getRelationTypeAndId(relation);
+    if (relationId === 'all') {
```

Comment:
I would recommend not to check the relation id. My suggestion is to extend the `RelationTypes` with `match_all` and `subcourse_all` and still provide the id of the match / subcourse. This allows the events to be assigned to the individual matches / subcourses later. We would therefore save `match_all/1` in the user_achievement instead of `match/all`.